pkg/api/validation/validation.go

```diff
@@ -121,7 +115,7 @@ func maskTrailingDash(name string) string {
 // ValidatePodName can be used to check whether the given pod name is valid.
 // Prefix indicates this name will be used as part of generation, in which case
 // trailing dashes are allowed.
-func ValidatePodName(name string, prefix bool) (bool, string) {
+func ValidatePodName(name string, prefix bool) (bool, []string) {
```

Comment:
thinking out loud-- `var ValidatePodName = NameIsDNSSubdomain` -- less code churn if something else like this changes?

(shrug)